packages/dd-trace/src/appsec/index.js

```diff
@@ -299,6 +301,7 @@ function disable () {
 
   // Channel#unsubscribe() is undefined for non active channels
   if (bodyParser.hasSubscribers) bodyParser.unsubscribe(onRequestBodyParsed)
+  if (multerParser.hasSubscribers) multerParser.unsubscribe(onRequestBodyParsed)
```

Comment:
should we check if body has been sent to the waf before in a previous `onRequestBodyParsed`? (to avoid sending the same body multiple times)

Reply:
could there be a case where multiple body middlewares parse it differently ? and thus we'd want to send them all to make sure we analyze all forms ?

Reply:
I think that we shouldn't. 

I'm thinking in juice-shop app... with `bodyParser.text()` middleware, and multer middleware. The good one is the last one, which will set the json object in `req.body`, we want to send the last to the waf.

Reply:
okay